src/components/Semester/Semester.vue

```diff
@@ -376,6 +398,18 @@ export default defineComponent({
       const courseCode = `${data.subject} ${data.catalogNbr}`;
       this.openConfirmationModal(`Added ${courseCode} to ${this.season} ${this.year}`);
     },
+    selectCourse(data: CornellCourseRosterCourse) {
+      // TODO @willespencer handle opening conflict modal better: should not happen if no conflicts
```

Comment:
Here, I would

1. find all the requirements this course would fulfill, using something like `computeGroupedRequirementFulfillmentReports()` (maybe refactoring some of that function)
2. use `getConstraintViolationsForSingleCourse()` to determine if there are any constraint violations for the course